### PR TITLE
Add Support for DD_TRACE_PARTIAL_FLUSH_ENABLED

### PR DESCRIPTION
**Pull request by sabrenner**
### What does this PR do?
Adds support for enabling/disabling partial flushing (which right now is always enabled by default via `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS`. This PR adds support via the environment variable `DD_TRACE_PARTIAL_FLUSH_ENABLED` and the config option `partialFlushEnabled`, with the config option taking precedence, as is standard for the tracer

### Motivation
Failing/flaky parametric tests in system tests. Also, gaining consistency across all language tracers. The default value of this variable is inconsistent across tracers, but is dependent upon implementation in each tracer. Since the Node.js tracer already enables partial flushing by default via `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS`, which defaults to `1000` when not specified in `process.env` or in config options, the default value of `DD_TRACE_PARTIAL_FLUSH_ENABLED` is `true`, but can now be explicitly disabled if partial flushing has an associated performance cost depending on use case.

### Additional Notes
WIP: need to add some documentation in the documentation repository.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -354,7 +354,15 @@ export declare interface TracerOptions {
   flushInterval?: number;
 
   /**
-   *  Number of spans before partially exporting a trace. This prevents keeping all the spans in memory for very large traces.
+   * Enables or disables large payloads being sent to APM.
```
Comment (sabrenner) [inline index.d.ts:357]: not sure about wording here, so any suggestions would be appreciated! Whatever is finalized for wording here is what I'll use for the documentation repo

File: packages/dd-trace/src/config.js [hunk 2]
```diff
@@ -253,6 +253,11 @@ class Config {
       process.env.DD_TRACE_AGENT_PROTOCOL_VERSION,
       '0.4'
     )
+    const DD_TRACE_PARTIAL_FLUSH_ENABLED = coalesce(
+      options.partialFlushEnabled,
```
Comment (sabrenner) [inline packages/dd-trace/src/config.js:257]: Also, any input on the name of this config options/the fact that this feature also lives under a config option is also appreciated

File: index.d.ts [hunk 3]
```diff
@@ -354,7 +354,15 @@ export declare interface TracerOptions {
...
+   * This prevents keeping all the spans in memory for very large traces.
+   * @default true
+   */
+  partialFlushEnabled?: boolean
```
Comment (rochdev) [inline index.d.ts:361]: Why would this ever be disabled? The only reason I could see would be to explicitly try to overload the agent, but that doesn't seem like a valid use case.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 3 above)
Comment (rochdev) [reply index.d.ts:361]: I see in the motivation a mention of system tests, but tests should test the behaviour of the library, the library shouldn't be implemented for tests.

Comment (sabrenner) [reply index.d.ts:361]: Yeah that's fair. I don't necessarily think this is needed either, but I put it up for review with the intention of consistency across tracers. If internal discussion between tracers leads to the conclusion that we don't need this change (or changes like this that, in general, conform with `system-tests`), I'm happy to close out this PR!

**Comment by sabrenner** [comment]
I'm closing this PR out for now due to differing internal opinion, and the lack of urgency for this implementation. If it resurfaces, this PR can be reopened and reevaluated.

**Pull request closed without merging**